src/renderer/inc/IRenderEngine.hpp

```diff
@@ -43,80 +43,50 @@ namespace Microsoft::Console::Render
             HyperlinkUnderline = 0x80
         };
 
-        virtual ~IRenderEngine() = 0;
+        constexpr IRenderEngine() = default;
+        virtual ~IRenderEngine(){};
+
+        IRenderEngine(const IRenderEngine&) = delete;
+        IRenderEngine& operator=(const IRenderEngine&) = delete;
 
-    protected:
-        IRenderEngine() = default;
-        IRenderEngine(const IRenderEngine&) = default;
         IRenderEngine(IRenderEngine&&) = default;
-        IRenderEngine& operator=(const IRenderEngine&) = default;
         IRenderEngine& operator=(IRenderEngine&&) = default;
 
-    public:
         [[nodiscard]] virtual HRESULT StartPaint() noexcept = 0;
         [[nodiscard]] virtual HRESULT EndPaint() noexcept = 0;
-

```

Comment:
Packing all these declaration together makes it more pleasant to view them in an editor.
It also simplifies copy/pasting them to fulfill the interfaces (since you need to copy/paste these into the implementation, remove `virtual` and replace `= 0` with `override`).